### PR TITLE
scripts: Update 'package.nls.json' automatically

### PR DESCRIPTION
**Pull request by JadenSimon**
## Problem
We have to update the 'package.nls.json' file manually after adding 'localize' calls (or when we change the text)

## Solution
Do it automatically apart of 'compile' (new script is 'generateNlsPackage'')
This script doesn't remove unused localized keys, though that can easily be changed.

Coming soon (maybe):
* Working localization 

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1504,11 +1504,12 @@
         "bundleDeps": "node ./build-scripts/bundleDeps.js",
         "clean": "node ./build-scripts/clean.js dist",
         "generateNonCodeFiles": "ts-node ./build-scripts/generateNonCodeFiles.ts",
+        "generateNLSPackage": "ts-node ./build-scripts/nlsHelper.ts",
```
Comment (justinmk3) [inline package.json:1507]: nit :)
```suggestion
        "generateNlsPackage": "ts-node ./build-scripts/nlsHelper.ts",
```

**Comment by bryceitoc9** [comment]
Were the package.nls.json changes made from running the script? If so, great! That being said, what happens if you have two keys in the code with different default strings?

File: build-scripts/nlsHelper.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+interface LocalizationElement {
+    messages: string[]
+    keys: string[]
+}
+
+const nlsdata = JSON.parse(fs.readFileSync('./dist/nls.metadata.json').toString())
+const outdata = JSON.parse(fs.readFileSync('package.nls.json').toString())
+
+// Reads the strings extracted by the nls webpack loader, then overwrites our package.nls.json data
```
Comment (justinmk3) [inline build-scripts/nlsHelper.ts:18]: Nice find! 💯

**Comment by JadenSimon** [comment]
Yeah the updated package is from the script. The code will take precedence over what's currently in the package.

**Review by justinmk3 — APPROVED** [review]

**Comment by JadenSimon** [comment]
Closing this in favor of a proper implementation of vscode-nls.

**Pull request closed without merging**